app/src/app.py

```diff
@@ -0,0 +1,87 @@
+import datetime
+import json
+import os
+import pathlib
+
+import boto3
+import hyp3_sdk
+import requests
+
+
+def get_granule_ur_pattern(granule_ur: str) -> str:
+    return granule_ur.rsplit('-', 1)[0] + '-*'
+
+
+def exists_in_cmr(cmr_domain: str, granule_ur: str) -> bool:
+    url = f'https://{cmr_domain}/search/granules.umm_json'
+    params = (
+        ('short_name', 'ARIA_S1_GUNW'),
+        ('granule_ur', get_granule_ur_pattern(granule_ur)),
+        ('options[granule_ur][pattern]', 'true'),
+        ('page_size', 1),
+    )
+    response = requests.get(url, params=params)
+    response.raise_for_status()
+    if response.json()['items']:
+        print(f'{granule_ur} already exists in CMR as {response.json()["items"][0]["umm"]["GranuleUR"]}')
```

Comment:
I might err on the side of explicitly checking that the returned granule UR matches our `granule_ur` parameter, but I'm less confident in my CMR query knowledge.

Reply:
We skip ingest if the granule is already in the archive, even if the existing granule has a different processor version. We're doing a wildcard search on granule UR, so the UR we get back from CMR might not exactly match the one for the candidate product:

`S1-GUNW-D-R-036-tops-20250131_20250119-041630-00025E_00035N-PP-0574-v3_0_1 already exists in CMR as S1-GUNW-D-R-036-tops-20250131_20250119-041630-00025E_00035N-PP-0574-v3_0_0`